lib/lrama/binding.rb

```diff
@@ -0,0 +1,23 @@
+module Lrama
+  class Binding
+    attr_reader :actual_args, :count
+
+    def initialize(parameters, actual_args)
+      @parameters = parameters
+      @actual_args = actual_args
+      @count = parameters.count
+      @parameter_to_arg = parameters.zip(actual_args).map do |param, arg|
+        [param.s_value, arg]
+      end.to_h
+    end
+
+    def resolve_symbol(symbol)
+      if symbol.is_a?(Lexer::Token::InstantiateRule)
+        symbol.args.map! { |arg| resolve_symbol(arg) }
```

Comment:
Is it safe to destruct `InstantiateRule#args` when `InstantiateRule` appears in ParameterizingRule, so that the `InstantiateRule` might be referred more than once?

Reply:
It might be better to build new `InstantiateRule` from `symbol` with replaced `args`.

Reply:
Indeed it is. I changed the `InstantiateRule` to rebuild.